resteasy-rxjava2/src/main/java/org/jboss/resteasy/rxjava2/SingleRxInvokerImpl.java

```diff
@@ -1,171 +1,189 @@
 package org.jboss.resteasy.rxjava2;
 
-import javax.ws.rs.client.CompletionStageRxInvoker;
 import javax.ws.rs.client.Entity;
 import javax.ws.rs.core.GenericType;
 import javax.ws.rs.core.Response;
 
+import io.reactivex.Flowable;
 import io.reactivex.Single;
+import org.jboss.resteasy.client.jaxrs.internal.ClientInvocationBuilder;
+import org.jboss.resteasy.client.jaxrs.internal.UniPublisherRxInvokerImpl;
+import org.reactivestreams.Publisher;
+
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
 
 @SuppressWarnings("unchecked")
 public class SingleRxInvokerImpl implements SingleRxInvoker
 {
-   private final CompletionStageRxInvoker completionStageRxInvoker;
-   private final SingleProvider singleProvider;
 
-   public SingleRxInvokerImpl(final CompletionStageRxInvoker completionStageRxInvoker)
+   private final SinglePublisherInvoker publisherInvoker;
+
+   public SingleRxInvokerImpl(final ClientInvocationBuilder builder)
+   {
+      publisherInvoker = new SinglePublisherInvoker(Objects.requireNonNull(builder));
+   }
+
+   static class SinglePublisherInvoker extends UniPublisherRxInvokerImpl
    {
-      this.completionStageRxInvoker = completionStageRxInvoker;
-      this.singleProvider = new SingleProvider();
+      SinglePublisherInvoker(final ClientInvocationBuilder builder)
+      {
+         super(builder);
+      }
+
+      @Override
+      protected <T> Publisher<T> toPublisher(CompletionStage<T> completable) {
+         return Flowable.fromFuture(completable.toCompletableFuture());
```

Comment:
Would it make sense to use Single.fromFuture() here?

Reply:
Sorry. I missed this one.  `Single#fromFuture` won't get us a `Publisher`.